x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -166,7 +166,7 @@ function isLoginAttemptWithProviderType(
  * @param sessionValue The session value to check.
  */
 function isLegacyProviderSession(sessionValue: any) {
-  return typeof sessionValue?.provider === 'string';
+  return typeof sessionValue?.provider?.name !== 'string';
 }
```

Comment:
A current (7.x) Kibana session looks like this:
```
{
   "state":{
      "authorization":"Basic ZWxhc3RpYzpjaGFuZ2VtZQ=="
   },
   "provider":{
      "type":"basic",
      "name":"basic"
   },
   "idleTimeoutExpiration":null,
   "lifespanExpiration":null,
   "path":"/"
}
```

A 6.8 Kibana session looks like this:
```
{
   "value":{
      "state":{
         "authorization":"Basic ZWxhc3RpYzpjaGFuZ2VtZQ=="
      },
      "provider":"basic"
   },
   "expires":null
}
```

This change will treat any session that doesnt have a `provider.name` string value as a legacy session, and subsequently discard it.